core/domain/exp_jobs_one_off.py

```diff
@@ -470,3 +471,115 @@ def map(item):
     @staticmethod
     def reduce(key, values):
         yield (key, values)
+
+
+class ExplorationContentValidationJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that checks the html content of an exploration and validates it.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+
+        def find(key, dictionary):
+            """Finds the value of the key inside all the nested dictionaries
+            in a given dictionary.
+
+            Args:
+                key: str. The key whose value is to be found.
+                dictionary: dict or list. The dictionary or list in which the
+                    key is to be searched.
+
+            Returns:
+                list. The values of the key in the given dictionary.
+            """
+            if isinstance(dictionary, list):
+                for d in dictionary:
+                    if isinstance(d, (dict, list)):
+                        for result in find(key, d):
+                            yield result
+            else:
+                for k, v in dictionary.iteritems():
+                    if k == key:
+                        yield v
+                    elif isinstance(v, dict):
+                        for result in find(key, v):
+                            yield result
+                    elif isinstance(v, list):
+                        for d in v:
+                            if isinstance(d, (dict, list)):
+                                for result in find(key, d):
+                                    yield result
+
+        if item.deleted:
+            return
+
+        exploration = exp_services.get_exploration_from_model(item)
+
+        allowed_parent_list = {
```

Comment:
Hmm I suggest we don't maintain the information of `allowed_parent_list` and `allowed_tag_list` in code. Instead, we can maintain it in a separate yaml/json file, and then this code can parse that file. This way, if we need to update the allowed parents and allowed tags, we don't need to modify the code itself (we can just update that file).

Which, I assume we'll need to do when we switch from textAngular->ckEditor anyways.

Reply:
Done! 
Though I think we may need to keep the `htmlValidation.yaml` file in some other directory(if required).